velox/exec/Driver.cpp

```diff
@@ -479,10 +479,19 @@ StopReason Driver::runInternal(
               }
               RuntimeStatWriterScopeGuard statsWriterGuard(op);
               if (op->isFinished()) {
-                auto timer =
-                    createDeltaCpuWallTimer([op](const CpuWallTiming& timing) {
-                      op->stats().wlock()->finishTiming.add(timing);
-                    });
+                auto setCpuTime = [op, nextOp](const CpuWallTiming& timing) {
+                  op->stats().wlock()->finishTiming.add(timing);
+                  // storing backgroundCpuTime separate and leave folding it
+                  // into finishTiming to compute framework (e.g., PrestoTask)
+                  const CpuWallTiming opBackgroundTiming{
+                      1, 0, op->getBackgroundCpuTimeMs()};
+                  op->stats().wlock()->backgroundTiming.add(opBackgroundTiming);
+                  const CpuWallTiming nextOpBackgroundTiming{
+                      1, 0, nextOp->getBackgroundCpuTimeMs()};
+                  nextOp->stats().wlock()->backgroundTiming.add(
+                      nextOpBackgroundTiming);
+                };
+                auto timer = createDeltaCpuWallTimer(std::move(setCpuTime));
                 RuntimeStatWriterScopeGuard statsWriterGuard(nextOp);
                 TestValue::adjust(
                     "facebook::velox::exec::Driver::runInternal::noMoreInput",
```

Comment:
Why we are interested on collecting background time on noMoreInput call? Thanks!

Reply:
Fixed